hivemq-edge/src/main/java/com/hivemq/persistence/domain/DomainTagPersistenceImpl.java

```diff
@@ -151,4 +153,13 @@ public DomainTagPersistenceImpl() {
             return List.of();
         }
     }
+
+    @Override
+    public @NotNull DomainTag getTag(@NotNull final String tagId) {
+        final DomainTag domainTag = nodeIdToDomainTag.get(tagId);
+        if (domainTag == null) {
+            throw new TagNotFoundException("Tag'" + tagId + "' was not found in the persistence.");
```

Comment:
Why is that an exception?
I'd rather return an optional since this looks like a valid case not an exceptional one.

Reply:
no it is absolutely not a valid case. This will hinder any operation of the adapter and is a severe error case as the system stops working as intended.